### ISSUE TITLE
Update documentation now that client connect has become implicit

### ISSUE THREAD

**Issue by brunchboy** [relayrides/pushy#565]
I’m working on helping update `pushy-clj` to the current API, from version 0.9.2, and have run into a few stumbling blocks where the documentation could better reflect the new way of doing things. The main project `README.clj` still shows the following example for constructing an `ApnsClient`:

```java
final ApnsClient apnsClient = new ApnsClientBuilder()
        .setClientCredentials(new File("/path/to/certificate.p12"), "p12-file-password")
        .build();
```

However, now that there is no longer an explicit `connect` call made to the client, the above code will actually result in an `IllegalStateException` because “No APNs server address specified.” It seems that `setApnsServer()` now needs to be called on the `ApnsClientBuilder` before `build()`.

The same seems to be true for the alternate, Token, configuration example which follows:

```java
final ApnsClient apnsClient = new ApnsClientBuilder()
        .setSigningKey(ApnsSigningKey.loadFromPkcs8File(new File("/path/to/key.p8"),
                "TEAMID1234", "KEYID67890"))
        .build();
```

Also, [line 49 of ApnsClientBuilder.java](https://github.com/relayrides/pushy/blob/release-0.11.3/pushy/src/main/java/com/turo/pushy/apns/ApnsClientBuilder.java#L49) states “All settings are optional.” This is no longer true, so it should probably say “All settings apart from the APNS Server and credentials are optional.”

Finally, you might want to mention this `IllegalStateException` in the [JavaDoc for the `build()` method](https://github.com/relayrides/pushy/blob/release-0.11.3/pushy/src/main/java/com/turo/pushy/apns/ApnsClientBuilder.java#L467-L475). It can be thrown if there is no server configured, if there are no credentials configured, or if both TLS  credentials and a signing key are configured.

**Comment by jchambers** [comment]
All legit criticism. Will try to get it sorted shortly. Thanks for the report!

**Comment by jchambers** [comment]
Fixed everything in 402661e. Thanks very much for the heads-up!
